sources/web/datalab/polymer/components/table-details/table-details.ts

```diff
@@ -108,6 +115,31 @@ class TableDetailsElement extends Polymer.Element {
     return table ? this._bytesToReadableSize(table.numLongTermBytes) : '';
   }
 
+  _flattenFields(fields: gapi.client.bigquery.Field[]) {
+    const flatFields: gapi.client.bigquery.Field[] = [];
+    fields.forEach((field) => {
+
+      // First push the record field itself
+      flatFields.push(field);
+
+      // Then flatten it and push its children
+      if (field.type === 'RECORD' && field.fields) {
+        // Make sure we copy the flattened nested fields before modifying their
+        // name to prepend the parent field name
+        const nestedFields = [...this._flattenFields(field.fields)];
```

Comment:
This looks like it recursively flattens, so if the table has something nested a few levels, these names could get pretty long. You know what would be nice to add? A resize bar between the results list and the preview pane so the user could make the preview pane wider in cases like this. In a separate PR, of course.